tests/waku_relay/test_protocol.nim

```diff
@@ -0,0 +1,1302 @@
+      let 
+        otherSwitch = newTestSwitch()
+        otherNode = await newTestWakuRelay(otherSwitch)
+
+      await allFutures(otherSwitch.start(), otherNode.start())
+      let otherRemotePeerInfo = otherSwitch.peerInfo.toRemotePeerInfo()
+      check await peerManager.connectRelay(otherRemotePeerInfo)
+
+      var otherHandlerFuture = newPushHandlerFuture()
+      proc otherSimpleFutureHandler(topic: PubsubTopic, message: WakuMessage) {.async, gcsafe.} =
+        otherHandlerFuture.complete((topic, message))
+
+      # When subscribing both nodes to the same Pubsub Topic
+      discard node.subscribe(pubsubTopic, simpleFutureHandler)
+      discard otherNode.subscribe(pubsubTopic, otherSimpleFutureHandler)
+
+      # Then both nodes are subscribed
+      check:
+        node.isSubscribed(pubsubTopic)
+        node.subscribedTopics == pubsubTopicSeq
+        otherNode.isSubscribed(pubsubTopic)
+        otherNode.subscribedTopics == pubsubTopicSeq
+
+      await sleepAsync(500.millis)
+
+      # When publishing a message in node
+      let fromOtherWakuMessage = fakeWakuMessage("fromOther")
+      discard await node.publish(pubsubTopic, fromOtherWakuMessage)
+
+      # Then the message is published in both nodes
+      check:
+        await handlerFuture.withTimeout(3.seconds)
+        await otherHandlerFuture.withTimeout(3.seconds)
+
+      let 
+        (topic1, message1) = handlerFuture.read()
+        (otherTopic1, otherMessage1) = otherHandlerFuture.read()
+      check:
+        topic1 == pubsubTopic
+        message1 == fromOtherWakuMessage
+        otherTopic1 == pubsubTopic
+        otherMessage1 == fromOtherWakuMessage
+      
+      # When publishing a message in the other node
+      handlerFuture = newPushHandlerFuture()
+      otherHandlerFuture = newPushHandlerFuture()
+      let fromNodeWakuMessage = fakeWakuMessage("fromNode")
+      discard await node.publish(pubsubTopic, fromNodeWakuMessage)
+      discard await otherNode.publish(pubsubTopic, fromNodeWakuMessage)
+
+      # Then the message is published only in the subscribed node
+      check:
+        await handlerFuture.withTimeout(3.seconds)
+        await otherHandlerFuture.withTimeout(3.seconds)
+      
+      let 
+        (topic2, message2) = handlerFuture.read()
+        (otherTopic2, otherMessage2) = otherHandlerFuture.read()
+      check:
+        topic2 == pubsubTopic
+        message2 == fromNodeWakuMessage
+        otherTopic2 == pubsubTopic
+        otherMessage2 == fromNodeWakuMessage
+      
+      # Finally stop the other node
+      await allFutures(otherSwitch.stop(), otherNode.stop())
+
+    asyncTest "Refreshing subscription":
+      # Given a subscribed node
+      node.subscribe(pubsubTopic, simpleFutureHandler)
+      check:
+        node.isSubscribed(pubsubTopic)
+        node.subscribedTopics == pubsubTopicSeq
+      let fromOtherWakuMessage = fakeWakuMessage("fromOther")
+      discard await node.publish(pubsubTopic, fromOtherWakuMessage)
```

Comment:
Shouldn't it be `fromNodeWakuMessage`? There's no other node in this test. Same for the other names in this test that use the word other.

Reply:
You're right, I don't know why I named it that way.